src/Overlay.tsx

```diff
@@ -0,0 +1,83 @@
+import styled from 'styled-components'
+import React, {ReactElement} from 'react'
+import {get, COMMON, POSITION} from './constants'
+import {ComponentProps} from './utils/types'
+import { useOverlay, AnchoredPositionHookSettings, TouchOrMouseEvent } from './hooks'
+import Portal from './Portal'
+
+type StyledOverlayProps = {
+  width?: keyof typeof widthMap
+  height?: keyof typeof heightMap
+  visibility: 'visible' | 'hidden'
+}
+
+const heightMap = {
+  'sm': '480px',
+  'md': '640px',
+  'auto': 'auto'
+}
+
+const widthMap = {
+  'sm': '256px',
+  'md': '320px',
+  'lg': '480px',
+  'xl': '640px',
+  'auto': 'auto'
+}
+
+const StyledOverlay  = styled.div<StyledOverlayProps>`
+  background-color: ${get('overlay.bg')};
+  box-shadow: ${get('overlay.popover.boxShadow')};
+  position: absolute;
+  z-index: 100;
+  min-width: 192px;
+  max-width: 480px;
+  height: ${props => heightMap[props.height || 'auto']};
+  width: ${props => widthMap[props.width || 'auto']};
+  border-radius: ${get('overlay.borderRadius')};
+  overflow: hidden;
+  animation: overlay-appear 200ms ${get('animation.easeOutCubic')};
+
+  @keyframes overlay-appear {
+    0% {
+      opacity: 0;
+      transform: translateY(${get('space.2')});
+    }
+    100% {
+      opacity: 1;
+    }
+  }
+  visibility: ${props => props.visibility};
+  ${COMMON};
+  ${POSITION};
+`
+
+export type OverlayProps = {
+  ignoreClickRefs: React.RefObject<HTMLElement> []
+  initialFocusRef?: React.RefObject<HTMLElement>
+  returnFocusRef: React.RefObject<HTMLElement>
+  anchorRef: React.RefObject<HTMLElement>
+  onClickOutside: (e: TouchOrMouseEvent) => void
+  onEscape: (e: KeyboardEvent) => void
```

Comment:
Since we don't manage the open state we can't close the `Overlay` ourselves when the user presses `Esc`. Not ideal, but I think it's okay for now.